spec.bs

```diff
@@ -1680,17 +1680,17 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
           1. Let |directFromSellerSignalsForBuyer| be the result of running
             [=get direct from seller signals for a buyer=] with |directFromSellerSignals|, and |ig|'s
             [=interest group/owner=].
-          1. Let |generatedBid| be the result of [=generate a bid=] given |allTrustedBiddingSignals|,
+          1. Let « |generatedBid|, |bidDebugReportInfo| » be the result of [=generate a bid=] given |allTrustedBiddingSignals|,
             |auctionSignals|, a [=map/clone=] of |browserSignals|, |perBuyerSignals|,
             |directFromSellerSignalsForBuyer|, |perBuyerTimeout|, |expectedCurrency|, |ig|, and
             |auctionStartTime|.
           1. Let |generateBidDuration| be the [=duration from=] |generateBidStartTime| to |settings|'s
             [=environment settings object/current monotonic time=], in milliseconds.
           1. If |perBuyerCumulativeTimeout| is not null, decrement |perBuyerCumulativeTimeout| by
             |generateBidDuration|.
-          1. If |generatedBid|'s [=generated bid/no bid=] is true:
-            1. [=Register a bid for forDebuggingOnly reports=] given |generatedBid| and
-               |generatedBidsForDebuggingReports|.
+          1. If |generatedBid| is null:

```

Comment:
Can generated bid be null? Generated bid is not a nullable type, and the `[=generate a bid=]` algorithm doesn't ever return null. In the failure case it returns `failure`, so I think that return value has to be handled explicitly itself.

Reply:
It can return it:
https://pr-preview.s3.amazonaws.com/morlovich/turtledove/pull/1089.html#evaluate-a-bidding-script:~:text=If%20generatedBid%20is%20a%20generated%20bid%20and%20generatedBid%E2%80%99s%20bid%27s%20value%20%E2%89%A4%200%2C%20then%20set%20generatedBid%20to%20null.

Reply:
... Before this change that check was followed by a condition that made a weirdo bid object with "no bid" set to true; see line 3439 on the pre-change side; in this revision it instead just returns null + a non-null bidDebugReportInfo object, since that's where all the useful stuff is.

(Next CL will end up returning an empty list of generated bids + bidDebugReportInfo, basically generalizing 0-1 generated bids to 0-n of them)

Reply:
Got it, thanks!